### PR TITLE
Setting SubscriptionID and TopicID in PubSub transport

### PR DESCRIPTION
**Pull request by nachocano**
- Adding a new options method to be able to set up subscription and topic id in the pubsub transport. This is needed to be able to load the topic from the transport context.

### CODE REVIEW

**Comment by n3wscott** [comment]
LgMT

**Comment by nachocano** [comment]
> LgMT

LgMT --> LGTM?

**Comment by nachocano** [comment]
any idea with the test fails? way to re-run?

**Comment by n3wscott** [comment]
Fast fingers... Lgtm

Try the tests locally

**Comment by nachocano** [comment]
@n3wscott  master is broken... See https://circleci.com/gh/cloudevents/sdk-go/370
It seems that we are not setting the datacontenttype anymore? I removed them from tests and it passes.

**Comment by n3wscott** [comment]
Master is not broken, it is _now_ working and the test is wrong.

File: pkg/binding/format/format_test.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func TestMarshalUnmarshal(t *testing.T) {
 	assert.NoError(e.SetData("foo"))
 	b, err := format.Marshal(format.JSON.MediaType(), e)
 	assert.NoError(err)
-	assert.Equal(`{"data":"foo","datacontenttype":"application/json","id":"id","source":"source","specversion":"0.3","type":"type"}`, string(b))
+	assert.Equal(`{"data":"foo","id":"id","source":"source","specversion":"0.3","type":"type"}`, string(b))
```
Comment (matzew) [inline pkg/binding/format/format_test.go]: @nachocano see this PR:

https://github.com/cloudevents/sdk-go/pull/235/files

(sorry, seeing this just now ..)

Comment (nachocano) [reply pkg/binding/format/format_test.go]: thanks @matzew , just rebased!

**Pull request merged**